### DISCUSSION TITLE
Welcome to csgo-modded-server Discussions!

### DISCUSSION THREAD

**Discussion by kus** [kus/cs2-modded-server#22] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by razorkillz** [comment]
I am trying to add a custom map to gungame. Im editing mapcycle_gg.txt which is in 
csgo-modded-server-master\server\csgo
how do i keep it from over writing ive been tyring to figure it out...i have it running on my actual computer and i have been trying to add maps to the server so what would i need to do to get them to work

**Reply by kus** [reply]
If you are on Linux use "[custom files](https://github.com/kus/csgo-modded-server#custom-files)". 

You would put it in `/custom_files/mapcycle_gg.txt` the contents of "custom_files" is merged into the csgo folder. So it overwrites any default file.

**Reply by razorkillz** [reply]
is it the same if im using windows?
edit: nvm i found it out...thx for helping

**Reply by kus** [reply]
@razorkillz I didn't have it for Windows, but I have updated the Windows setup to now include it.

You can download the 2 updated files [win.bat](https://raw.githubusercontent.com/kus/csgo-modded-server/master/win.bat) and [win.ini](https://raw.githubusercontent.com/kus/csgo-modded-server/master/win.ini) and overwrite them in your current installation.

Then you would put `mapcycle_gg.txt` in `custom_files/mapcycle_gg.txt`.

To confirm you did it correctly, when you run `win.bat` you should see `Copying custom files from "custom_files".` and then you can confirm by looking at `server/csgo/mapcycle_gg.txt` and it should have your custom `mapcycle_gg.txt`.

**Comment by IdreesAbuEtewy** [comment]
hello my friend , i have a question 
what is the mod or method that provide the multimod within one server to choose between from admin menu like (retake , arena ,surf ) and each plugin or mod have its own config

**Reply by kus** [reply]
I don't fully understand your question, but if you're asking how to change mods read [this section](https://github.com/kus/csgo-modded-server#acessing-admin-menu).

If you are asking how I make a "multimod" server on one server, it's the way I've set up the server files - no special mod that does it. Load the server with just SourceMod and no plugins. Then execute a config (mod) to load all the plugins for that mod. Then if you want to change mods it unloads all the plugins and loads the new ones. I've been maintaining this mod for 4 years and it's been ongoing, so lots of little things I've had to tweak over that time.

**Reply by IdreesAbuEtewy** [reply]
thanks for the replay

**Comment by IdreesAbuEtewy** [comment]
thanks man for this masterpiece plugins , its help me alot , but if i want suggest to add zombie and jail brake to it will be fantastic

**Comment by IdreesAbuEtewy** [comment]
Im using your repo . but i want install zombie reloaded mod too , i try everything , but it crash every time warmup end , have you success install it before ? or its cant be in multi mod server , duo conflict in gamedata ?

**Reply by kus** [reply]
Give me a link to zombie mod and jail break and if it is straightforward I'll add them.

**Reply by IdreesAbuEtewy** [reply]
For zombie i found theses two repo is latest version one by Francisco  https://github.com/Franc1sco/sm-zombiereloaded-3-Franug-Edition
and this one latest https://github.com/Game-Host-Bros/mod-zombie-escape/  ,  for jailbreak didnt install it yet but i found this as the latest one https://github.com/eyal282/csgo-jailbreak-package/tree/master

**Reply by kus** [reply]
@DaBiQ I have added both [Zombie:Reloaded](https://forums.alliedmods.net/showthread.php?t=277597) and [MyJailbreak](https://forums.alliedmods.net/showthread.php?t=283212).

The Jailbreak mod you linked didn't work and was very new. I used [MyJailbreak](https://forums.alliedmods.net/showthread.php?t=283212) which is built on top of [SM_Hosties v2](https://forums.alliedmods.net/showthread.php?t=108810).

You can access both mods via the admin menu. I recommend restarting the server in-between changing mods (you can do this from the admin menu and reconnect in 10 seconds).

If you want to make any customisations you can do that by adding "[custom files](https://github.com/kus/csgo-modded-server#custom-files)" which will overwrite my defaults.

**Reply by IdreesAbuEtewy** [reply]
thanks , appreciate it , i will try it right away

**Reply by IdreesAbuEtewy** [reply]
im on windows and in start.bat should i add key_api to be able to download workshop map?

**Reply by kus** [reply]
I just updated the repo to include the api key in the Windows start script.

Follow [these instructions](https://github.com/kus/csgo-modded-server#running-on-windows) to get it setup for windows.

Register a [Steam Web API Key here](https://steamcommunity.com/dev/apikey).

Edit `win.ini` and put the `Steam Web API Key` in `csgo_api_key`.

**Comment by dvgmdvgm** [comment]
@kus hello. Just install your server and at this step "To check everything is working correctly"
But console output looks like that —
```
meta list
Listing 1 plugin:
  [01] <NOFILE>
```
and
```
css_plugins list
Unknown command 'css_plugins'!
```
So something went wrong but i don't know what. Can u help me>

**Reply by kus** [reply]
I need more information:

- Where are you installing the server? Local machine, vm, online, what host?
- OS
- Output of server console when you start the server
- Environment variables used and how you are starting the server

**Comment by NitroDenko097** [comment]
hi, I have a problem starting the server because I have these things here that I don't know where they should be inserted, I have a Windows 10 virtual machine. The things I need are below 
:start

:: Deleting addons folder so no old plugins are left to cause issues
:: If you have modifications in your addons/ folder they should be in custom_files as these are merged at the end
echo Deleting addons folder.
rmdir /S /Q "%ROOT_DIR%server\game\csgo\addons\"

:: Patch server with mod files
echo Copying mod files.
xcopy "%ROOT_DIR%game\csgo\*" "%ROOT_DIR%server\game\csgo\" /K /S /E /I /H /Y >NUL

:: Merge Windows specific files
echo Merging Windows specific files.
xcopy "%ROOT_DIR%game\csgo\addons\windows\*" "%ROOT_DIR%server\game\csgo\addons\" /K /S /E /I /H /Y >NUL

:: Merge your custom files in
echo Copying custom files from "%custom_folder%".
xcopy "%ROOT_DIR%%custom_folder%\*" "%ROOT_DIR%server\game\csgo\" /K /S /E /I /H /Y >NUL

in cmd it gives me these errors

File not found - *
Merging Windows specific files.
File not found - *
Copying custom files from "custom_files".
CS2 started.

**Comment by allice-flp** [comment]
> To check everything is working correctly run the following commands in the server console:
> 
> meta list and you should see CounterStrikeSharp in the output
> css_plugins list and you should see a few plugins in the output

done, but there is no output. we've done everything, sitting here for ours but we have no output.
you dont mention what we could do if there is no output, is there any possibility?

running on a new linux system, trying to run a training-server for our team

**Reply by allice-flp** [reply]
we fixed almost everything, but we cant change the admins.
admins.json is in custom-files but it still get changed every time we start, i dont get it, never was so confused

**Comment by Kju1911** [comment]
hey, me and my friend are trying to run your server on linux. I setted all correctly but after all no one can connect. I cant connect even in LAN mode. We relly need help to set the server up
